sema4ai-http-helper/src/sema4ai_http/__init__.py

```diff
@@ -58,20 +58,8 @@ class _NetworkConfig:
     def __init__(self) -> None:
         self.connection_pool = self._build_connection_pool()
 
-    @staticmethod
-    def _get_network_settings_path() -> Path:
-        if sys.platform == "win32":
-            localappdata = os.environ.get("LOCALAPPDATA")
-            if not localappdata:
-                raise RuntimeError("Error. LOCALAPPDATA not defined in environment!")
-            sema4_home = Path(localappdata) / "sema4ai"
-        else:
-            # Linux/Mac
-            sema4_home = Path("~/.sema4ai").expanduser()
-
-        return sema4_home / "network-settings.yaml"
-
-    def _get_current_config_profile(self) -> dict:
+    @cached_property
+    def profile_config(self) -> dict:
```

Comment:
Do we know the expected shape of the dict here (maybe a TypedDict would be better if we know about it).